src/openforms/submissions/models.py

```diff
@@ -62,10 +61,20 @@ def get_next_step(self) -> Optional["SubmissionStep"]:
         candidates = (
             step
             for step in self.submission_steps[offset:]
-            if not step.completed and step.available
+            if not step.completed and step.is_applicable
         )
         return next(candidates, None)
 
+    def get_submission_step(self, form_step_uuid: str) -> "SubmissionStep":
```

Comment:
```suggestion
    def get_submission_step(self, form_step_uuid: str) -> Optional["SubmissionStep"]:
```
if there's a chance that `None` is returned, type hint should indicate this.